Cython/Shadow.py

```diff
@@ -337,7 +337,7 @@ def __init__(self, cast_from=_Unspecified, **data):
             setattr(self, key, value)
 
     def __setattr__(self, key, value):
-        if key in '__dict__':
+        if key == '__dict__':
             CythonType.__setattr__(self, key, value)
```

Comment:
I think this probably meant to say `if key in self.__dict__`.

Reply:
I thought this was probably right - this code looks necessary for line 343 to work

Reply:
Ah, yes, right. That's why it's there.

Reply:
Suggests to me that this is untested code, though. We should test the assignment in `tests/run/pure.py`.

Reply:
Sorry, I should probably have added more details but it seemed self-explanatory to me. Basically, that condition should only evaluate to `True` when assigning `__dict__`, but it currently does for `_`, `d`, `i`, etc. as well resulting into the following potential issues;

- Non-member are being assigned to the object instead of raising.
- The one-field rule can be bypassed.
- Valid members that pass the condition are being assigned raw and are never cast to the specified type.

As for the tests, I think the assignment is being indirectly tested through:

https://github.com/cython/cython/blob/9cff140813b0ede6d107ed7031052513df251ed9/tests/run/pure.pyx#L93-L105

But work as intended because `n` is not contained into `__dict__`. Renaming `n` to, for example, `d` in that routine on the master branch would likely result into the third point mentioned above because it would be assigned and returned as-is instead of being properly cast to the defined type.